cpp/src/io/parquet/page_enc.cu

```diff
@@ -1613,6 +1613,51 @@ dremel_data get_dremel_data(column_view h_col,
     return std::make_tuple(std::move(empties), std::move(empties_idx), empties_size);
   };
 
+  // Check if there are empty lists with empty offsets in this column
+  bool has_empty_list_offsets = false;
+  {
+    auto curr_col = h_col;
+    while (is_nested(curr_col.type())) {
+      if (curr_col.type().id() == type_id::LIST) {
+        auto lcv = lists_column_view(curr_col);
+        if (lcv.offsets().size() == 0) {
+          has_empty_list_offsets = true;
+          break;
+        }
+        curr_col = lcv.child();
+      } else if (curr_col.type().id() == type_id::STRUCT) {
+        curr_col = curr_col.child(0);
+      }
+    }
+  }
+  std::unique_ptr<column> empty_list_offset_col;
+  if (has_empty_list_offsets) {
+    empty_list_offset_col = make_fixed_width_column(data_type(type_id::INT32), 1);
+    cudaMemsetAsync(empty_list_offset_col->mutable_view().head(), 0, sizeof(size_type), stream);
```

Comment:
I wonder if this could be done with a simple `column_view` wrapper around a pointer to 
`__constant__ int32_t empty_list_offset_data = 0;`
This is probably not worth doing for a single value just to save the Memset call.